Modix/Modules/CatModule.cs

```diff
@@ -0,0 +1,178 @@
+﻿namespace Modix.Modules
+{
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Discord.Commands;
+    using Newtonsoft.Json;
+    using Serilog;
+
+    public class CatModule : ModuleBase
+    {
+        // Url for cat
+        public struct URL
+        {
+            public string file { get; set; }
+        }
+
+        private static readonly HttpClient Client = new HttpClient();
+
+        [Command("cat"), Summary("Gets a cat")]
+        public async Task Cat(string gif = "")
+        {
+            // 30 seconds to find a cat
+            var cts = new CancellationTokenSource(30000);
+            var token = cts.Token;
+
+            // Regular picture
+            if (gif.Equals(string.Empty))
+            {
+                await GetCatPicture(token);
+            }
+            else if (gif.Equals("gif"))
+            {
+                // Gif cat
+                await GetCatGif(token);
+            }
+            else
+            {
+                await Context.Channel.SendMessageAsync("Use `!cat` or `!cat gif`");
+            }
+        }
+
+        private async Task GetCatGif(CancellationToken token)
+        {
+            var obj = new URL();
+            var json = string.Empty;
+            var flag = false;
+
+            // five chances to find a cat gif
+            for (var i = 0; i < 5; i++)
+            {
+                json = await DownloadCatJson(token);
+
+                // Check and make sure the string isn't empty before attempting to deserialize cat
+                if (string.IsNullOrWhiteSpace(json))
+                {
+                    Log.Error("Empty JSON String");
```

Comment:
Error level log events will log to discord as well. Only use if something is actually broken (and include more detail)